cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/event/EventBus.java

```diff
@@ -88,9 +105,16 @@ public void register(Object object) {
         do {
             for (Method m : listenerClass.getDeclaredMethods()) {
                 if (m.isAnnotationPresent(com.cloudant.sync.event.Subscribe.class)) {
+
+                    if (!Modifier.isPublic(m.getModifiers())){
+                        throw new IllegalArgumentException("Subscriber method with @Subscribe is required to be public");
+                    }
+
                     Class[] params = m.getParameterTypes();
                     if (params.length == 1) {
                         methods.add(new SubscriberMethod(m, params[0]));
+                    } else {
+                        throw new IllegalArgumentException("Subscriber method is required to have only 1 parameter");
```

Comment:
Should we consider doing additional checks here that the parameter is one of the notification classes. I believe they are all in the same package now so we could check that maybe?